agent/php_curl_md.c

```diff
@@ -42,76 +35,142 @@ static void nr_php_curl_md_destroy(nr_php_curl_md_t* metadata) {
-    nr_php_curl_md_t* __metadata = NULL;                                  \
-                                                                          \
-    ensure_curl_metadata_hashmap();                                       \
-                                                                          \
-    __metadata = nr_hashmap_index_get(NRTXNGLOBAL(curl_metadata), __id);  \
-    if (!__metadata) {                                                    \
-      __metadata = nr_zalloc(sizeof(nr_php_curl_md_t));                   \
-      nr_hashmap_index_set(NRTXNGLOBAL(curl_metadata), __id, __metadata); \
-    }                                                                     \
-                                                                          \
-    __metadata;                                                           \
-  })
+static void ensure_curl_multi_metadata_hashmap()
+{
+  if (!NRTXNGLOBAL(curl_multi_metadata)) {
+    NRTXNGLOBAL(curl_multi_metadata) = nr_hashmap_create(
+        (nr_hashmap_dtor_func_t)nr_php_curl_multi_md_destroy);
+  }
+}
 
-#define get_curl_multi_metadata(mh)                                     \
-  ({                                                                    \
-    uint64_t __id = (uint64_t)nr_php_zval_resource_id(mh);              \
-    nr_php_curl_multi_md_t* __multi_metadata;                           \
-    size_t async_index;                                                 \
-                                                                        \
-    ensure_curl_multi_metadata_hashmap();                               \
-                                                                        \
-    __multi_metadata                                                    \
-        = nr_hashmap_index_get(NRTXNGLOBAL(curl_multi_metadata), __id); \
-                                                                        \
-    if (!__multi_metadata) {                                            \
-      __multi_metadata = nr_zalloc(sizeof(nr_php_curl_multi_md_t));     \
-      nr_hashmap_index_set(NRTXNGLOBAL(curl_multi_metadata), __id,      \
-                           __multi_metadata);                           \
-      async_index = nr_hashmap_count(NRTXNGLOBAL(curl_multi_metadata)); \
-      if (!nr_php_curl_multi_md_init(__multi_metadata, async_index)) {  \
-        nr_free(__multi_metadata);                                      \
-      }                                                                 \
-    }                                                                   \
-    __multi_metadata;                                                   \
-  })
+static nr_php_curl_md_t* get_curl_metadata(const zval* ch TSRMLS_DC)
+{
+#if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO /* PHP 8.0+ */
+    uint64_t id = (uint64_t)nr_php_zval_object_id(ch);
+#else
+    uint64_t id = (uint64_t)nr_php_zval_resource_id(ch);
+#endif
+    if (id == 0)
```

Comment:
Flip around for `yoda` conditions ([styleguide](https://github.com/newrelic/newrelic-php-agent/blob/main/STYLEGUIDE.md)).

Same comment for [here](https://github.com/newrelic/newrelic-php-agent/pull/318/files#diff-6340c4a86ab2af8769a24114fe8dc973281e36f3b5d98aab7cb2a16834eca284R124).